content/en/blog/2019/isolated-clusters/index.md

```diff
@@ -0,0 +1,1828 @@
+among system components using distinct encryption keys.
+{{< /quote >}}
+
+Various compliance standards recommend isolating environments that process sensitive data from the rest of the
+organization.
+The [Payment Card Industry (PCI) Data Security Standard](https://www.pcisecuritystandards.org/pci_security/)
+recommends implementing network isolation for _cardholder data_ environment and requires isolating this environment from
+the [DMZ](https://en.wikipedia.org/wiki/DMZ_(computing)).
+[FedRAMP Authorization Boundary Guidance](https://www.fedramp.gov/assets/resources/documents/CSP_A_FedRAMP_Authorization_Boundary_Guidance.pdf)
+describes _authorization boundary_ for federal information and data, while
+[NIST Special Publication 800-37, Revision 2, Risk Management Framework for Information Systems and Organizations: A System Life Cycle Approach for Security and Privacy](https://doi.org/10.6028/NIST.SP.800-37r2)
+recommends protecting of such a boundary in _Appendix G, Authorization Boundary Considerations_:
+
+{{< quote >}}
+Dividing a system into subsystems (i.e., divide and conquer) facilitates a targeted application of controls to achieve
+adequate security, protection of individual privacy, and a cost-effective risk management process. Dividing complex
+systems into subsystems also supports the important security concepts of domain separation and network segmentation,
+which can be significant when dealing with high value assets. When systems are divided into subsystems, organizations
+may choose to develop individual subsystem security and privacy plans or address the system and subsystems in the same
+security and privacy plans.
+Information security and privacy architectures play a key part in the process of dividing complex systems into
+subsystems. This includes monitoring and controlling communications at internal boundaries among subsystems and
+selecting, allocating, and implementing controls that meet or exceed the security and privacy requirements of the
+constituent subsystems.
+{{< /quote >}}
+
+The approach I propose in this blog post facilitates division of a system into subsystems with different security and
+compliance requirements, and facilitates the boundary protection.
+You put each subsystem into a separate Kubernetes cluster, preferably on a separate network. You install
+a dedicated Istio control plane into each cluster and connect the Istio meshes using ingress and egress gateways.
+The gateways monitor and control cross-cluster traffic at the boundary of each cluster.
+
+## Features of the proposed approach
+
+The approach I propose has the following features:
+
+- **non-uniform naming**. The `withdraw` service in the `accounts` namespace in one cluster might have
+different functionality and API than the `withdraw` services in the `accounts` namespace in other clusters. The same
+service with the same functionality can have different names and can reside in different namespaces in different
```

Comment:
I propose to proceed in an agile manner: to publish a blog post about a concrete use case with a concrete implementation.
 
> So we can't talk about something like that in a blog post since we haven't agreed that we want to officially support that. 

Why? This is what blog posts are good for. This is not an official task or concept, it is an opinionated blog post. Note that this blog post uses Istio AS IS, it just shows possible usage of Istio to address a particular use case.
 
> It can introduce a lot of potential problems that I don't think we've fully thought through.

I am sorry, I do not think this is the right way to proceed. This is too waterfall, not agile at all. How can we think fully about some architecture pattern without seeing a single example of it? Why shouldn't we publish a blog post and collect user's feedback on that?

Reply:
We don't want to tell users about a pattern that we're not supporting. Please remove any discussion about a single service using different names in different clusters from this blog post.